server/server.go

```diff
@@ -834,6 +838,35 @@ func (s *Server) createClient(conn net.Conn) *client {
 	return c
 }
 
+// This will save off a closed client in a ring buffer such that
+// /connz can inspect. Useful for debugging, etc.
+func (s *Server) saveClosedClient(c *client, nc net.Conn, reason ClosedState) {
```

Comment:
This could be a receiving of the ring buffer object (especially if we use a dedicated lock there)

Reply:
Yes it could. Feels like a server verb though that just uses the ring buffer. We do the time stamp etc.